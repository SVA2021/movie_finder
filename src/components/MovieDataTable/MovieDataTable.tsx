import {Paper, Table, TableBody, TableCell, TableContainer, TableRow, Typography} from "@mui/material";
import {FC, memo} from 'react';
import {TFullCard} from "../../features/movieFinder/movieFinderTypes";
import {getTimeFromMinutes} from "../../utils";

interface MovieDataTableProps {
  movie: TFullCard | null
}

export const MovieDataTable: FC<MovieDataTableProps> = memo(({movie}) => {

  const createData = (name: string, description: string | number,) => ({name, description});

  const rows = [
    createData('Время', getTimeFromMinutes(movie?.filmLength)),
    createData('Страны', movie?.countries.map((v) => v.country).join(', ') ?? ' - '),
    createData('Год', movie?.year ?? ' - '),
    createData('Жанр', movie?.genres.map((v) => v.genre).join(', ') ?? ' - '),
    createData('', movie?.description ?? ' - '),
  ]

  return (
    <TableContainer component={Paper}>
      <Table aria-label="основная информация" sx={{maxWidth: '100%'}} >
        <TableBody>
          {rows.map((row) => (
            <TableRow key={row.name}>
              <TableCell component="th" scope="row" align={'left'} padding={'none'} sx={{border: 'none'}} >
                <Typography variant="body1">
                  {row.name}
                </Typography>
              </TableCell>
              <TableCell align={'left'} sx={{border: 'none'}}>
                <Typography variant="body1">
                  {row.description}
                </Typography>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
});