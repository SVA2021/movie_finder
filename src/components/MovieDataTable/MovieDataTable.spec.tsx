import {render, screen} from "@testing-library/react";
import {MovieDataTable} from "./MovieDataTable";

describe('render MovieDataTable', () => {

  const setUp = () => render(<MovieDataTable movie={null} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('movie-data-table')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/Время/i)).toBeInTheDocument();
    expect(screen.getByText(/Страны/i)).toBeInTheDocument();
    expect(screen.getByText(/Год/i)).toBeInTheDocument();
    expect(screen.getByText(/Жанр/i)).toBeInTheDocument();
  });

})