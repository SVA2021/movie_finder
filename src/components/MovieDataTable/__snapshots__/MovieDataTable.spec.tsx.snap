// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieDataTable snapshot test 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  data-testid="movie-data-table"
>
  <table
    aria-label="основная информация"
    class="MuiTable-root css-fj9a4a-MuiTable-root"
  >
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-6u6txs-MuiTableCell-root"
          scope="row"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Время
          </p>
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-96k1qc-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          />
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-6u6txs-MuiTableCell-root"
          scope="row"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Страны
          </p>
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-96k1qc-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             - 
          </p>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-6u6txs-MuiTableCell-root"
          scope="row"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Год
          </p>
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-96k1qc-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             - 
          </p>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-6u6txs-MuiTableCell-root"
          scope="row"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Жанр
          </p>
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-96k1qc-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             - 
          </p>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-6u6txs-MuiTableCell-root"
          scope="row"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          />
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-96k1qc-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             - 
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
